airbyte-commons-worker/src/main/java/io/airbyte/workers/normalization/NormalizationRunnerFactory.java

```diff
@@ -38,12 +41,22 @@ public class NormalizationRunnerFactory {
 
   public static NormalizationRunner create(final String connectorImageName,
                                            final ProcessFactory processFactory,
-                                           final String normalizationVersion) {
+                                           final String normalizationVersion,
+                                           final String normalizationImage) {
     final var valuePair = getNormalizationInfoForConnector(connectorImageName);
+    final String factoryNormalizationImage = String.format("%s:%s", valuePair.getLeft(), normalizationVersion);
+    log.error("destination normalization runner factory -> {}", factoryNormalizationImage);
+    if (Objects.nonNull(normalizationImage)
+        && !normalizationImage.equalsIgnoreCase(factoryNormalizationImage)) {
+      log.error(
+          "The normalization image name or tag in the definition file is different from the normalization image or tag in the NormalizationRunnerFactory!");
+      log.error(
+          "the definition file value - {}, the NormalizationRunnerFactory value - {}", normalizationImage, factoryNormalizationImage);
+    }
     return new DefaultNormalizationRunner(
         valuePair.getRight(),
         processFactory,
-        String.format("%s:%s", valuePair.getLeft(), normalizationVersion));
+        factoryNormalizationImage);
```

Comment:
Yes, so far I've left using the image from the platform and so far I'm just logging the difference. In the next pull request, **NormalizationRunnerFactory** will be removed along with the create method, and instead, we will directly use the new constructor for `DefaultNormalizationRunner`. (instead of `NormalizationRunnerFactory.create(......)` -> `new DefaultNormalizationRunner(processFactory, normalizationImage)`). and I added the **factoryNormalizationImage** variable to the method so as not to use the String.format(...) construction several times =)